pkg/auth/handler/oauth/userinfo.go

```diff
@@ -30,13 +32,17 @@ type UserInfoHandler struct {
 	Logger           UserInfoHandlerLogger
 	Database         *appdb.Handle
 	UserInfoProvider ProtocolUserInfoProvider
+	OAuth            *config.OAuthConfig
 }
 
 func (h *UserInfoHandler) ServeHTTP(rw http.ResponseWriter, r *http.Request) {
 	s := session.GetSession(r.Context())
+	scopes := oauth.SessionScopes(s)
+	clientID := oauth.ClientID(s)
+	client, _ := h.OAuth.GetClient(clientID)
```

Comment:
client could also be nil if the third party client is deleted. We now assume a nil client is IDP session. If the client is deleted, we treat it as third party with no scopes.

Reply:
The session should become invalid if the client is deleted, so I didn't handle this case.

Updated to introduce `ClientLike` to handle the case :)